src/base/ddl_log.c

```diff
@@ -290,69 +297,99 @@ logddl_set_broker_info (const int index, const char *br_name)
+	  memcpy (ddl_audit_handle.sql_text, sql_text, ddl_audit_handle.alloc_size);
+
+	  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 5] = ' ';
+	  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 4] = '.';
+	  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 3] = '.';
+	  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 2] = '.';
+	  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 1] = '\0';
+	  ddl_audit_handle.sql_text_len = ddl_audit_handle.alloc_size - 1;
+	  return;
+	}
+
+      if (ddl_audit_handle.sql_text != ddl_audit_handle.zsql_text)
 	{
-	  if (ddl_audit_handle.alloc_size <= len)
+	  if (ddl_audit_handle.sql_text_len > 0)
 	    {
-	      int new_size = ddl_audit_handle.alloc_size;
-	      char *ptr;
+	      memcpy (ptr, ddl_audit_handle.sql_text, ddl_audit_handle.sql_text_len);
+	      ptr[ddl_audit_handle.sql_text_len] = '\0';
+	    }
+	  FREE_MEM (ddl_audit_handle.sql_text);
+	}
 
-	      while (new_size < len)
-		{
-		  new_size += SQL_TEXT_INCR;
-		}
+      ddl_audit_handle.sql_text = ptr;
+      ddl_audit_handle.alloc_size = new_size;
+    }
 
-	      ptr = (char *) MALLOC (new_size + 1);
-	      if (!ptr)
-		{		/* It's a failure, but let's copy as much space as we can */
-		  memcpy (ddl_audit_handle.sql_text, sql_text, ddl_audit_handle.alloc_size);
-
-		  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 5] = ' ';
-		  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 4] = '.';
-		  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 3] = '.';
-		  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 2] = '.';
-		  ddl_audit_handle.sql_text[ddl_audit_handle.alloc_size - 1] = '\0';
-		  return;
-		}
+  if (ddl_audit_handle.sql_text_len > 0)
+    {
+      memcpy (ddl_audit_handle.sql_text + ddl_audit_handle.sql_text_len, delim_str, delim_len);
+      ddl_audit_handle.sql_text_len += delim_len;
+    }
 
-	      if (ddl_audit_handle.sql_text != ddl_audit_handle.zsql_text)
-		{
-		  FREE_MEM (ddl_audit_handle.sql_text);
-		}
+  if (has_password_type == false)
+    {
+      memcpy (ddl_audit_handle.sql_text + ddl_audit_handle.sql_text_len, sql_text, sql_len);
+      ddl_audit_handle.sql_text_len += sql_len;
+    }
+  else
+    {
+      char chBk = sql_text[sql_len];
 
-	      ddl_audit_handle.sql_text = ptr;
-	      ddl_audit_handle.alloc_size = new_size;
-	    }
+      sql_text[sql_len] = '\0';
+      tlen =
+	password_snprint (ddl_audit_handle.sql_text + ddl_audit_handle.sql_text_len,
+			  (ddl_audit_handle.alloc_size - ddl_audit_handle.sql_text_len), sql_text, hide_pwd_info_ptr);
+      sql_text[sql_len] = chBk;
 
-	  memcpy (ddl_audit_handle.sql_text, sql_text, len);
-	  ddl_audit_handle.sql_text[len] = '\0';
-	}
+      ddl_audit_handle.sql_text_len += tlen;
     }
-}
 
+  ddl_audit_handle.sql_text[ddl_audit_handle.sql_text_len] = '\0';
+}
 
 bool
-logddl_set_stmt_type (int stmt_type)
+logddl_set_stmt_type_internal (int stmt_type, PT_NODE * statement)
```

Comment:
In general, xxx_internal () is a static function and is usually called from the xxx () function. And, xxx() can be called from a function outside that file. Defining only xxx_internal() without the function xxx() seems a bit strange to me.

Reply:
OK, I agree.
I will fix it.   And logddl_set_sql_text_internal() also.